src/abstractdataframe/show.jl

```diff
@@ -597,50 +597,64 @@ function _show(io::IO,
                rowlabel::Symbol = :Row,
                summary::Bool = true,
                eltypes::Bool = true,
-               rowid=nothing,
-               truncstring::Int)
+               truncstring::Int = 32,
+               kwargs...)
+
     _check_consistency(df)
 
-    # we will pass around this buffer to avoid its reallocation in ourstrwidth
-    buffer = IOBuffer(Vector{UInt8}(undef, 80), read=true, write=true)
+    names = permutedims(propertynames(df))
```

Comment:
just for me to to know - why do you need them to be `1 x ncol(df)` matrices? It seems a bit unintuitive

Reply:
Because we will have a header (variable names) and a sub-header (variable types). PrettyTables uses a matrix to print this. The first line of the matrix is the header, the others are the sub-headers. So, to use `vcat`, I need to transpose it first.